localstack/services/apigateway/helpers.py

```diff
@@ -619,7 +618,8 @@ def host_based_url(rest_api_id: str, path: str, stage_name: str = None):
 
 def get_execute_api_endpoint(api_id: str, protocol: str = "") -> str:
     port = config.get_edge_port_http()
```

Comment:
Can we change this to localstack_host.host_and_port instead of using get_edge_port_http?

Reply:
mhm, maybe that's a sign to remove `get_edge_port_http()` anyways with 3.0 and force extensions to update 🤔 
It's a bit short notice and uncool that all extensions need to be re-released but it might be nicer to break it with a major change ...
I could tackle it in a follow up if we wanna go this router 🎢

Reply:
I agree we should use `localstack_host().host_and_port()` in this case 👍